### ISSUE TITLE
Refactor/fix kubeconfig management in `opsctl`

### ISSUE THREAD

**Issue by anvddriesch** [giantswarm/roadmap#1657]
Related https://github.com/giantswarm/roadmap/issues/1577
 
Kubeconfig management in `opsctl` has grown over time and by now is quite convoluted and hard to follow.
Furthermore, with CAPI installations not running vault and client certificate creation therefore not being possible, some commands fail.
Let's have a thorough look at the current state, do a refactor and fix the broken commands.

**Comment by erkanerol** [comment]
`opsctl create etcdbackup` doesn't work for CAPI clusters because of this reason. 

```
➜ opsctl create etcdbackup -i guppy --workloadcluster etcd1
Error: Invalid config: k8srestconfig.Config.Address must not be empty when not using k8srestconfig.Config.InCluster or k8srestconfig.Config.KubeConfig
	/home/runner/go/pkg/mod/github.com/giantswarm/k8sclient/v7@v7.0.1/pkg/k8srestconfig/rest_config.go:108
	/home/runner/work/opsctl/opsctl/pkg/k8sclient/k8sclient.go:158
	/home/runner/work/opsctl/opsctl/command/create/etcdbackup/command.go:201
	/home/runner/work/opsctl/opsctl/command/create/etcdbackup/command.go:124
	/home/runner/work/opsctl/opsctl/main.go:57

To increase verbosity, re-run with --level=debug
```

and we don't have `--kubeconfig` flag for this command, which is the workaround for `opsctl deploy` command. Should I create a separate issue for this?

**Comment by marians** [comment]
We talked about this in the team and think we can work on it after current projects, ~ in a few weeks.

**Comment by weatherhog** [comment]
- [ ] refactor how kubeconfigs are managed in `opsctl`

**Comment by gawertm** [comment]
revisit this after cert management for capi is done (potentially including vault)

**Comment by gawertm** [comment]
@anvddriesch do you believe this is relevant enough that someone would work on it anytime soon? otherwise I#d say we can close it

**Comment by anvddriesch** [comment]
I can't speak to the etcd backups but I know that for most cases where broken kubeconfig was a problem people have added other methods. (For example, we now use kubectl gs update app instead of opsctl deploy)
